File: app/src/main/java/team/triplog/presentation/adapter/MainTripAdapter.java
package team.triplog.presentation.adapter;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import team.triplog.presentation.fragment.MainTripLogFragment;
import team.triplog.presentation.fragment.MainTripPlanFragment;

public class MainTripAdapter extends FragmentStatePagerAdapter {

    int tabCount;

    MainTripAdapter(FragmentManager fm, int tabCount) {
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
        this.tabCount = tabCount;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new MainTripPlanFragment();
            case 1:
                return new MainTripLogFragment();
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return tabCount;
    }
}

File: app/src/main/java/team/triplog/presentation/adapter/MainTripAdapter.kt
package team.triplog.presentation.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import team.triplog.presentation.fragment.MainTripLogFragment
import team.triplog.presentation.fragment.MainTripPlanFragment

class MainTripAdapter(
    fm: FragmentManager?,
    var tabCount: Int
) : FragmentStatePagerAdapter(
    fm!!, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT
) {
    override fun getCount(): Int = tabCount

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> MainTripPlanFragment()
            else -> MainTripLogFragment()
        }
    }
}